### PR TITLE
TE: Self Serve Unified Replay Flow

### PR DESCRIPTION
**Pull request by cecilynie**
Unify the selfserve replay flow. Modify replay to take clone function(inactive function), make sure replay to take historical anomalies, create endpoint to enable copy from cloned function configuration in total to original function, create endpoint to copy anomaly history in time range, enable delete anomaly history in time range. 

The whole flow has been tested by multiple angularities and multiple metrics

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java [hunk 1]
```diff
@@ -92,9 +94,63 @@ public Long cloneFunctionGetId(@PathParam("id") Long id,
     }
   }
 
+  /**
+   * Copy the Identity of source anomaly function (with srcId) over in total to destination anomaly function (with destId)
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: Identity is a vague term here. Please clarify the identity here.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java [hunk 2]
```diff
@@ -92,9 +94,63 @@ public Long cloneFunctionGetId(@PathParam("id") Long id,
...
+   * This in fact updates source anomaly function's properties into destination function's properties
+   * @param srcId : the source function with Identity to be copied to
+   * @param destId : the destination function Id that have properties to be copied to source function
+   * @return OK is success
+   */
+  @POST
+  @Path("function/{srcId}/copyIdentity/{destId}")
+  public Response copyIdentity(@PathParam("srcId") @NotNull Long srcId,
+      @PathParam("destId") @NotNull Long destId) {
+    AnomalyFunctionDTO srcAnomalyFunction = anomalyFunctionDAO.findById(srcId);
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: Please check if both functions exist.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java [hunk 3]
```diff
@@ -92,9 +94,63 @@ public Long cloneFunctionGetId(@PathParam("id") Long id,
...
+   * After "copyIdentity", the two functions will become:
+   *        (srcId, srcFunctionName, destProperties)
+   *        (destId, destFunctionName, destProperties)
+   * This in fact updates source anomaly function's properties into destination function's properties
+   * @param srcId : the source function with Identity to be copied to
+   * @param destId : the destination function Id that have properties to be copied to source function
+   * @return OK is success
+   */
+  @POST
+  @Path("function/{srcId}/copyIdentity/{destId}")
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: Refer this to @bl44 or @puneetjaiswal. Please make sure the path is consistent with all the others.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java [hunk 4]
```diff
@@ -92,9 +94,63 @@ public Long cloneFunctionGetId(@PathParam("id") Long id,
...
+  @Path("function/{srcId}/cloneAnomalies/{destId}")
+  public Response ClonedAnomalies(@PathParam("srcId") @NotNull long srcId,
+      @PathParam("destId") @NotNull long destId,
+      @QueryParam("start") String startTimeIso,
+      @QueryParam("end") String endTimeIso) throws Exception{
+
+    long start = 0;
+    long end = DateTime.now().getMillis();
 
+    if (startTimeIso != null) {
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: If the string is invalid, users are not notified. Please send proper error massage to user.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java [hunk 5]
```diff
@@ -282,8 +352,8 @@ public Boolean cloneAnomalyInstances(Long srcId, Long destId) {
    * @param monitoringWindowEndTime The start time of the monitoring window (in milli-second)
    */
   public Map<String, Integer> deleteExistingAnomalies(long functionId,
-      long monitoringWindowStartTime,
-      long monitoringWindowEndTime) {
+      Long monitoringWindowStartTime,
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: Please also document the null condition in Java Doc.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java (hunk 1 above)
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: Thanks, fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java (hunk 2 above)
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java (hunk 4 above)
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java (hunk 5 above)
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: removed this part, force function to take not null arguments

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java [hunk 6]
```diff
@@ -92,10 +94,80 @@ public Long cloneFunctionGetId(@PathParam("id") Long id,
...
+  @POST
+  @Path("function/{srcId}/copyIdentity/{destId}")
+  public Response copyIdentity(@PathParam("srcId") @NotNull Long srcId,
+      @PathParam("destId") @NotNull Long destId) {
+    AnomalyFunctionDTO srcAnomalyFunction = anomalyFunctionDAO.findById(srcId);
+    AnomalyFunctionDTO destAnomalyFunction = anomalyFunctionDAO.findById(destId);
+    if (srcAnomalyFunction == null) {
+      // LOG and exit
+      LOG.error("Anomaly Function With id [{}] does not found", srcId);
+      return null;
```
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java]: fixed, thanks for your advice!

**Pull request merged**